docs/API/V1/contents.md

```diff
@@ -0,0 +1,108 @@
+# `GET /api/v1/datasets/contents/<dataset><path>`
+
+This API returns an `application/json` document describing the contents of the referenced relative `<path>` within the `<dataset>` tarball representation.
+
+## URI parameters
+
+`<dataset>` string \
+The resource ID of a Pbench dataset on the server.
+
+`<path>`    string \
+The relative resource path of an item in the dataset inventory, as captured by the Pbench agent packaging; for example, `/` for the root, or `/1-default/` for the default first iteration directory.
+
+## Request headers
+
+`authorization: bearer` token \
+*Bearer* schema authorization is required to access any non-public dataset. E.g., `authorization: Bearer <token>`
+
+## Response headers
+
+`content-type: application/json` \
+The return is a serialized JSON object with information about the named file.
+
+## Response status
+
+`401`   **UNAUTHORIZED** \
+The client did not provide an authentication token and there is no public dataset with the name `<dataset>`.
```

Comment:
This means there *is* no authentication; disambiguating the expiration case depends on the client knowing whether it supplied authentication. If not, then it's an unauthenticated client looking for data that requires authorization; if it did, then it's reasonably safe to infer that the token has expired.

Even the API implementation doesn't really know, because the Flask auth decorators have detected the expiration and invalidated/removed the token before we get to our API code.

I suppose I could add that to the documentation, though; it's a useful detail, but also potentially a bit confounding. 😆